recipes/cairo/meson/conanfile.py

```diff
@@ -224,166 +226,119 @@ def package(self):
+                add_component_and_base_requirements("cairo-glesv3", [], ["GLESv2"])
+            elif self.options.with_opengl == "gles2":
+                add_component_and_base_requirements("cairo-glesv2", [], ["GLESv2"])
+            if self.settings.os in ["Linux", "FreeBSD"]:
+                add_component_and_base_requirements("cairo-egl", ["egl::egl"])
 
         if self.options.with_zlib:
-            self.cpp_info.components["cairo-pdf"].names["pkg_config"] = "cairo-pdf"
-            self.cpp_info.components["cairo-pdf"].requires = ["cairo_", "zlib::zlib"]
+            add_component_and_base_requirements("cairo-script", ["zlib::zlib"])
+            add_component_and_base_requirements("cairo-ps", ["zlib::zlib"])
+            add_component_and_base_requirements("cairo-pdf", ["zlib::zlib"])
+            self.cpp_info.components["cairo-script-interpreter"].set_property("pkg_config_name", "cairo-script-interpreter")
+            self.cpp_info.components["cairo-script-interpreter"].libs = ["cairo-script-interpreter"]
+            self.cpp_info.components["cairo-script-interpreter"].requires = ["cairo_"]
 
-        if self.options.with_zlib and self.options.with_png:
-            self.cpp_info.components["cairo-xml"].names["pkg_config"] = "cairo-xml"
-            self.cpp_info.components["cairo-xml"].requires = ["cairo_", "zlib::zlib"]
+            if self.options.with_png:
+                add_component_and_base_requirements("cairo-xml", ["zlib::zlib"])
+                add_component_and_base_requirements("cairo-util_", ["expat::expat"])
 
         if self.options.tee:
-            self.cpp_info.components["cairo-tee"].names["pkg_config"] = "cairo-tee"
+            self.cpp_info.components["cairo-tee"].set_property("pkg_config_name", "cairo-tee")
             self.cpp_info.components["cairo-tee"].requires = ["cairo_"]
 
         # util directory
         if self.options.with_glib:
-            self.cpp_info.components["cairo-gobject"].names["pkg_config"] = "cairo-gobject"
+            self.cpp_info.components["cairo-gobject"].set_property("pkg_config_name", "cairo-gobject")
             self.cpp_info.components["cairo-gobject"].libs = ["cairo-gobject"]
             self.cpp_info.components["cairo-gobject"].requires = ["cairo_", "glib::gobject-2.0", "glib::glib-2.0"]
 
-        if self.options.with_zlib:
-            self.cpp_info.components["cairo-script-interpreter"].libs = ["cairo-script-interpreter"]
-            self.cpp_info.components["cairo-script-interpreter"].requires = ["cairo_"]
-
-        # binary tools
-        if self.options.with_zlib and self.options.with_png:
-            self.cpp_info.components["cairo_util_"].requires.append("expat::expat") # trace-to-xml.c, xml-to-trace.c
+        self.cpp_info.components["cairo_"].requires += list(base_requirements)
+        self.cpp_info.components["cairo_"].system_libs += list(base_system_libs)
 
     def package_id(self):
         self.info.requires["glib"].full_package_mode()
```

Comment:
While we're at it, maybe change this to:
```python
if not self.options.shared:
         self.info.requires["glib"].full_package_mode()
```
In the default cause (shared glib), I don't see a reason for using `full_package_mode`.